src/App.jsx

```diff
@@ -14,33 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const SORT_BY_ALPHABET = 'Alphabetically';
+  const SORT_BY_LENGTH = 'ByLength';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sort = typeOfSort => {
+    if (typeOfSort === SORT_BY_ALPHABET) {
+      const sortedGoods = [...goods].sort((a, b) => {
+        return isReversed ? b.localeCompare(a) : a.localeCompare(b);
+      });
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      setGoods(sortedGoods);
+      setActiveSort(SORT_BY_ALPHABET);
+    } else {
+      const sortedGoods = [...goods].sort((a, b) => {
+        return isReversed ? b.length - a.length : a.length - b.length;
+      });
+
+      setGoods(sortedGoods);
+      setActiveSort(SORT_BY_LENGTH);
+    }
+  };
+
+  const reverseGoods = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort('');
+  };
+
+  const shouldShowReset = activeSort !== '' || isReversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': activeSort !== SORT_BY_ALPHABET,
+          })}
+          onClick={() => sort(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': activeSort !== SORT_BY_LENGTH,
+          })}
+          onClick={() => sort(SORT_BY_LENGTH)}
```

Comment:
Make a function that will call another one with the parameter you need
```suggestion
          onClick={sortByAlphabet}
        >
          Sort alphabetically
        </button>

        <button
          type="button"
          className={classNames('button is-success', {
            'is-light': activeSort !== SORT_BY_LENGTH,
          })}
          onClick={sortByLength}
```